src/ksym.rs

```diff
@@ -91,29 +95,25 @@ impl KSymResolver {
         }
     }
 
-    pub fn find_addresses_ksym(&self, addr: u64) -> Option<&Ksym> {
+    pub fn find_addresses_ksym(&self, addr: u64) -> impl Iterator<Item = &Ksym> {
         let mut l = 0;
         let mut r = self.syms.len();
 
-        if !self.syms.is_empty() && self.syms[0].addr > addr {
-            return None;
-        }
+        while l < r {
+            let m = (l + r) / 2;
+            let sym = &self.syms[m];
 
-        while l < (r - 1) {
-            let v = (l + r) / 2;
-            let sym = &self.syms[v];
-
-            if sym.addr == addr {
-                return Some(sym);
-            }
             if addr < sym.addr {
-                r = v;
+                r = m;
             } else {
-                l = v;
+                l = m + 1;
             }
         }
-
-        Some(&self.syms[l])
+        // `self.syms[l - 1].addr <= addr < self.syms[l].addr`.
```

Comment:
Nit: Consider removing this line? When I converted it into an assertion said assertion failed with an out of bounds access, so it does not seem kosher as-is. It also raises the question why it's not an assertion and what it really is conveying, given that it's *seemingly* valid Rust code but not actually. I feel like if it is an invariant we could express it in prose or use `debug_assert`.

Reply:
Translated to debug_assert().